### DISCUSSION TITLE
jsonschema draft 3

### DISCUSSION THREAD

**Discussion by mskaif** [koxudaxi/datamodel-code-generator#1389] [Q&A]
I have a schema that look like this:

`{
  "meta1": "desc",
  "meta2": "desc",
  "schemas": {
    "FirstObj": {
      "type": "object",
      "properties": {
        "prop1": {
          "type": "string",
          "description": "desc.",
          "format": "int32"
        },
        "prop2": {
          "$ref": "another_obj_In_the_same_file",
          "description": "desc."
        }
      },
      "description": "desc.",
      "id": "FirstObj"
    },
    "SecondObj": {
` etc...

This is jsonschema draft 3 compliant but what can I do to make it work with the current datamodel-code-generator? I need to create the models from the schema but exporting objects to separate json files before creating the classes is not convenient. How can I get the ref to work from one document? Please note that jsonschema document does not list the objects in the right order. I would really appreciate your help!

**Comment by koxudaxi** [comment]
@mskaif 
I'm sorry for my late reply, and I don't understand your question.
Can you provide the example expected and actual output?
If you want to separate the models by the file, add the namespace as a prefix to the model name.
https://github.com/koxudaxi/datamodel-code-generator/blob/188b904861cf3e800d7295190f3c551cb57989aa/tests/data/openapi/modular.yaml#L102

https://github.com/koxudaxi/datamodel-code-generator/blob/188b904861cf3e800d7295190f3c551cb57989aa/tests/data/expected/main/main_modular/models.py#L13
